cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -321,14 +349,21 @@ public void deleted(Id.Stream streamId) {
      * @return A {@link FileWriter} for writing {@link StreamEvent} to the given stream
      * @throws IOException if failed to create the file writer
      */
-    FileWriter<StreamEvent> create(Id.Stream streamId) throws Exception {
+    private FileWriter<StreamEvent> create(Id.Stream streamId) throws IOException {
```

Comment:
To do so, instead of using Callable, introduce a new interface, say `Task<V, E extends Exception>`, with a method `V perform() throws E`. Then the doAs signature becomes `<V, E extends Exception> doAs(NamespaceId I'd, Task<V, E> task)`